base_layer/key_manager/src/key_manager_service/interface.rs

```diff
@@ -56,6 +56,12 @@ pub enum AddResult {
     AlreadyExists,
 }
 
+#[derive(Debug, Eq, PartialEq)]
+pub struct KeyAndId<PK> {
+    pub key: PK,
+    pub key_id: KeyId<PK>,
+}
```

Comment:
key does not map to public key
```suggestion
pub struct KeyAndId<PK> {
    pub pub_key: PK,
    pub key_id: KeyId<PK>,
}
```